python/lsst/ip/diffim/detectAndMeasure.py

```diff
@@ -365,6 +364,31 @@ def run(self, science, matchedTemplate, difference,
                                    positiveFootprints=positives,
                                    negativeFootprints=negatives)
 
+    def _prepareInputs(self, difference):
+        """Ensure that we start with an empty detection and deblended mask.
+
+        Parameters
+        ----------
+        difference : `lsst.afw.image.ExposureF`
+            The difference image that will be used for detecting diaSources.
+            The mask plane will be modified in place.
+
+        Raises
+        ------
+        pipeBase.UpstreamFailureNoWorkFound
+            If the PSF is not usable for measurement
+        """
+        # Ensure that we start with an empty detection and deblended mask.
+        mask = difference.mask
+        for mp in self.config.clearMaskPlanes:
+            if mp not in mask.getMaskPlaneDict():
+                mask.addMaskPlane(mp)
+        mask &= ~mask.getPlaneBitMask(self.config.clearMaskPlanes)
+        # Check that we have a valid PSF now before we do more work
+        sigma = difference.psf.computeShape(difference.psf.getAveragePosition()).getDeterminantRadius()
+        if np.isnan(sigma):
+            raise pipeBase.UpstreamFailureNoWorkFound("Invalid PSF detected! PSF width evaluates to NaN.")
```

Comment:
I'd put this as the first line of the method, since there's no point in doing the rest if this is raised.